### PR TITLE
[Merged by Bors] - fix readme

### PR DESCRIPTION
**Pull request by ConvallariaMaj**
## Motivation

Env variables some times are not set properly, which affects tests and building. 

## Description

Fixed Readme to avoid problem described above

## Test Plan

- 

## TODO

- [ ] Explain motivation or link existing issue(s)
- [] Test changes and document test plan
- [ ] Update documentation as needed
- [ ] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,7 +97,7 @@ Read [this](https://github.com/golang/go/wiki/Modules#how-to-install-and-activat
 
 ### Setting Up Local Dev Environment
 
-Building is supported on OS X, Linux, FreeBSD, and Windows.
+Building is supported on OS X, Linux, FreeBSD, and Windows, which use GLIBC 2.34+ version
```
Comment (ivan4th) [inline README.md]:
```suggestion
Building is supported on OS X, Linux, FreeBSD, and Windows

In the case of Linux, GLIBC 2.34+ is required.
```

File: README.md [hunk 2]
```diff
@@ -112,6 +112,24 @@ Before building we need to set up the golang environment. Do this by running:
 make install
 ```
 
+Ensure, that environment is set correctly:
```
Comment (ivan4th) [inline README.md]:
```suggestion
Make sure the environment is set up correctly
```

File: README.md [hunk 3]
```diff
@@ -112,6 +112,24 @@ Before building we need to set up the golang environment. Do this by running:
...
+
+```bash
+make go-env-test
+```
+
+**CGO_CFLAGS** must be set to "-I<absolute_path_to_repo>/go-spacemesh/build/ -DSQLITE_ENABLE_DBSTAT_VTAB=1"
+**CGO_LDFLAGS** must be set to "-L<absolute_path_to_repo>/go-spacemesh/build/ -Wl,-rpath,$ORIGIN -Wl,-rpath,<absolute_path_to_repo>/go-spacemesh/build/"
+
+Ensure you have installed **OpenCL** library
```
Comment (ivan4th) [inline README.md]:
```suggestion
Make sure you have **OpenCL** library installed
```
but also we probably want to have version specified

File: README.md [hunk 4]
```diff
@@ -112,6 +112,24 @@ Before building we need to set up the golang environment. Do this by running:
...
+
+To check if setup was configured successfully, try to run: 
+```bash
+make test
+```
+
+You should not see compilation errors, but running test will take some time.
```
Comment (ivan4th) [inline README.md]:
```suggestion
There shouldn't be any build errors, but please note that running the tests will take some time.
```

**Review by ivan4th — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (poszu) [reply README.md]: Both "make sure" and "ensure" are correct.

**Review by ivan4th — APPROVED** [review]

**Comment by ConvallariaMaj** [comment]
bors merge

File: README.md [hunk 5]
```diff
@@ -97,7 +97,8 @@ Read [this](https://github.com/golang/go/wiki/Modules#how-to-install-and-activat
 
 ### Setting Up Local Dev Environment
 
-Building is supported on OS X, Linux, FreeBSD, and Windows.
+Building is supported on OS X, Linux, FreeBSD, and Windows
+In the case of Linux, GLIBC 2.34+ is required.
```
Comment (fasmat) [inline README.md]: More specifically we support

- Ubuntu 22.04 and newer (but any distro with GLIBC 2.34+ should work)
- MacOS 13 (Intel) and MacOS 14 (Arm) and newer
- Windows 10 and newer
- FreeBSD is not officially supported

**Review by fasmat — CHANGES_REQUESTED** [review]
The local dev environment isn't fully correct.

**Review by fasmat — APPROVED** [review]

**Comment by ConvallariaMaj** [comment]
bors merge

**Pull request closed without merging**